### ISSUE TITLE
Scala Repo Server: race condition if starting >1 instances?

### ISSUE THREAD

**Issue by alexanderdean** [snowplow/iglu#127]
If you start >1 instances, do all instances try to create the new apikeys table at the same time?

**Comment by fblundun** [comment]
@alexanderdean you're right - there are two potential race conditions [here](https://github.com/snowplow/iglu/blob/85b39b609de7175a039a3f8428340fdbb50c8b7c/2-repositories/scala-repo-server/src/main/scala/com.snowplowanalytics.iglu.server/Core.scala#L56-L61). Trying to create the table when it already exists will cause the application to terminate.

There's probably some sort of solution involving SQL locks. Alternatively we could catch any PSQLException whose message contains "already exists".

**Comment by alexanderdean** [comment]
Yes, we should probably just catch the exception, on the assumption that another server has beaten us to it. The failure to create because it already exists is benign...
